mm2src/mm2_metrics/src/mm_metrics.rs

```diff
@@ -285,7 +284,7 @@ pub mod prometheus {
         });
 
         let server = Server::try_bind(&address)
-            .map_err(|e| e.to_string())?
+            .map_err(|e| MmMetricsError::Internal(e.to_string()))?
```

Comment:
You're right that it's the internal mm2 error that we couldn't open the socket, but I think it would be more informative if we map the error to a `MmMetricsError::PrometheusServerError` or `MmMetricsError::PrometheusTransport`, or another error type.

Reply:
`MmMetricsError::PrometheusServerError` relates more